### ISSUE TITLE
Inform BFF of succesful authentication via non-default authentication scheme.

### ISSUE THREAD

**Issue by eugene-rebedailo** [DuendeSoftware/Support#546]
**Which version of Duende BFF are you using?**
v2.0.0

**Which version of .NET are you using?**
.NET 7.0

**Describe the question**
This issue relates to my previous question #520.

Having implemented authentication handler for a custom grant type in IdentityServer, I am struggling to connect the BFF and the authentication response - I was wondering about the way that BFF registers the successful authentications when they happen through regular OIDC and if I am somehow able to inform BFF the of my custom authentication handler.

I searched through the source code of BFF a little bit but its very distributed in the way it handles the bits and pieces of sign-in logic so it was hard to follow without an explanation.

I have implemented my custom grant type flow in "HandleAuthenticateAsync" - the acquisition of tokens and their validation, but I see that BFF is calling "ChallengeAsync" on the context in a lot of places.
I also was unable to find BFF subscriptions to any of the OIDC events like "OnTokenValidated" that are emitted by the OidcHandler.

So would you be able to clarify the potential ways to inform the BFF of my successful authentication and sign the user in with the default "cookie" scheme?

Side-note:
Not all of our applications are relying on the BFF, but I would like to reuse the custom authentication handler in those applications too, since this grant type is going to be used there too.
As such, I would like to avoid references to BFF inside the handler if possible.

**Comment by josephdecock** [comment]
The BFF is fairly decoupled from the implementation details of the authentication handler because of the way that the authentication service registers handlers as schemes. When the BFF calls ChallengeAsync, that invokes the default challenge authentication scheme. You can set that scheme on the authorization service to be your custom handler.

In a typical BFF, you would configure the schemes like this:
```
            services.AddAuthentication(options =>
                {
                    options.DefaultScheme = "cookie";
                    options.DefaultChallengeScheme = "oidc";
                    options.DefaultSignOutScheme = "oidc";
                })
                .AddOpenIdConnect("oidc", options =>
                {
                // etc
```

In your case, since you're not using the open id connect handler, you'll need to register your custom handler instead of openidconnect.

Probably implementing your handler as a RemoteAuthenticationHandler is the way to go, because it already understands that aftetr remote authentication, the user will need to be signed in with a different scheme - it gives you the [SignInScheme ](https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Core/src/RemoteAuthenticationHandler.cs#L28).
